diff -u a/node.def b/node.def
--- a/node.def	2023-01-10 22:31:03.213543174 +0000
+++ b/node.def	2023-01-10 22:25:55.977325048 +0000
@@ -1,6 +1,7 @@
 multi:
-type: ipv4net
+type: txt
 val_help: ipv4net; IPv4 address and prefix length
+val_help: ipv6net; IPv6 address and prefix length
 help: IP address
 
 syntax:expression: exec "/opt/vyatta/sbin/valid_address $VAR(@)"
diff -u a/vpn-config.pl b/vpn-config.pl
--- a/vpn-config.pl	2023-01-10 22:31:03.165541906 +0000
+++ b/vpn-config.pl	2023-01-10 22:25:55.977325048 +0000
@@ -476,12 +476,6 @@
             #
             if (defined($lip)) {
                 if ($lip eq 'any' || $lip eq 'default') {
-                    if ($isVti) {
-                        vpn_die(
-                            ["vpn", "ipsec", "site-to-site", "peer", $peer, $tunKeyword, "local-address"],
-                            "$vpn_cfg_err The local interface must be specified " . "for peer \"$peer\" $tunKeyword.\n"
-                        );
-                    }
                     if ($lip eq 'default') {
                         $genout .= "\tleft=%defaultroute\n";
                     } else {
@@ -544,7 +538,7 @@
             #
             my $leftsubnet = $vcVPN->returnValue("ipsec site-to-site peer $peer $tunKeyword local prefix");
             if ((defined($leftsubnet) && $leftsubnet eq 'any') || $isVti == 1) {
-                $leftsubnet = '0.0.0.0/0';
+                $leftsubnet = $isVti == 1 ? '0.0.0.0/0,::/0' : '0.0.0.0/0';
             }
 
             if (defined($leftsubnet)) {
@@ -632,7 +626,7 @@
             } else {
                 $rightsubnet = $remotesubnet;
                 if (defined($rightsubnet) && $rightsubnet eq 'any') {
-                    $rightsubnet = '0.0.0.0/0';
+                    $rightsubnet = $isVti == 1 ? '0.0.0.0/0,::/0' : '0.0.0.0/0';
                 }
             }
             if (defined($rightsubnet)) {
@@ -1002,7 +996,7 @@
                 my $index2 = (defined($authremoteid)) ? "$authremoteid" : $right;
                 if (  ($lip eq 'any' || $lip eq 'default' ||
                           ($key_exchange eq 'ikev1' and $index2 eq '%any')
-                      ) && !defined($dhcp_iface)) {
+                      ) && !defined($authid) && !defined($dhcp_iface)) {
                     if ($index1 =~ m/^@/) {
                         # In main mode PSK, the responder needs to look up the secret
                         # before the Peer's ID payload has been decoded, so the ID used
@@ -1119,7 +1113,11 @@
             # and up/down script hook.
             #
             if ($isVti) {
-                my $mark = vtiIntf::isVtimarkpresent($peer, $lip);
+                my $marklip = $lip;
+                if (defined($dhcp_iface) || $lip eq 'any' || $lip eq 'default') {
+                    $marklip = '0.0.0.0';
+                }
+                my $mark = vtiIntf::isVtimarkpresent($peer, $marklip);
                 if (!defined($mark) || $mark eq '') {
                     vpn_die(["vpn", "ipsec", "site-to-site", "peer", $peer, "vti"], "$vpn_cfg_err No mark specified for peer \"$peer\" vti\n");
                 } else {
diff -u a/vtiIntf.pm b/vtiIntf.pm
--- a/vtiIntf.pm	2023-01-10 22:31:03.201542858 +0000
+++ b/vtiIntf.pm	2024-01-20 20:46:57.832462021 +0000
@@ -69,6 +69,9 @@
     if ($tunop =~ m/local ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)/) {
         $local = $1;
     }
+    if ($tunop =~ m/local any/) {
+        $local = '0.0.0.0';
+    }
     if ($tunop =~ m/key ([0-9\.]+)/) {
         $mark = $1;
     }
@@ -84,6 +87,9 @@
     if ($key =~ m/local ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)/) {
         $local = $1;
     }
+    if ($key =~ m/local any/) {
+        $local = '0.0.0.0';
+    }
     return($remote, $local);
 }
 
diff -u a/vyatta-interfaces.pl b/vyatta-interfaces.pl
--- a/vyatta-interfaces.pl	2023-01-10 22:31:03.217543279 +0000
+++ b/vyatta-interfaces.pl	2023-01-10 22:25:55.977325048 +0000
@@ -357,10 +357,13 @@
 	next if ($addr eq 'dhcpv6');
 	if ($addr eq 'dhcp') {
 	    $dhcp = 1;
-  } elsif (!Vyatta::Interface::is_uniq_address($addr)) {
-    my $h = Vyatta::Misc::get_ipnet_intf_hash();
-            print "Error: duplicate address $addr on $h->{$addr}\n";
-            exit 1;
+	    next;
+	}
+	my $is_single = is_ipv4($addr) ? substr($addr, -3) eq '/32' : substr($addr, -4) eq '/128';
+	if (!$is_single && !Vyatta::Interface::is_uniq_address($addr)) {
+	    my $h = Vyatta::Misc::get_ipnet_intf_hash();
+	    print "Error: duplicate address $addr on $h->{$addr}\n";
+	    exit 1;
 	} elsif ( is_ipv4($addr) ) {
 	    $static_v4 = 1;
         }
diff -u a/vyatta-vti-config.pl b/vyatta-vti-config.pl
--- a/vyatta-vti-config.pl	2023-01-10 22:31:03.181542329 +0000
+++ b/vyatta-vti-config.pl	2023-01-10 22:25:55.977325048 +0000
@@ -116,8 +116,10 @@
     foreach my $peer (@peers) {
         my $lip = $vcVPN->returnValue("ipsec site-to-site peer $peer local-address");
 
-        # Do not support dhcp-interface option
-        next if (!defined($lip));
+        # Support for dhcp-interface option
+        if (!defined($lip) || $lip eq 'any' || $lip eq 'default') {
+            $lip = '0.0.0.0';
+        }
 
         my $tunName = $vcVPN->returnValue("ipsec site-to-site peer $peer vti bind");
 
@@ -157,6 +159,9 @@
         #
         my $mark;
         my $lip = $vcVPN->returnValue("ipsec site-to-site peer $peer local-address");
+        if (!defined($lip) || $lip eq 'default' || $lip eq 'any') {
+            $lip = '0.0.0.0';
+        }
         my $tunName = $vcVPN->returnValue("ipsec site-to-site peer $peer vti bind");
         my $change = 0;
 
@@ -170,7 +175,7 @@
             exit -1;
         }
         if (!(validateType('ipv4', $lip, 'quiet') || 
-              validateType('ipv6', $lip, 'quiet')) || ($lip eq '0.0.0.0')) {
+              validateType('ipv6', $lip, 'quiet'))) {
             print STDERR "$vti_cfg_err Invalid local-address \"$lip\", an ip address must be specified for VTIs.\n";
             exit -1;
         }
